src/main/java/org/terasology/caves/CaveFacetProvider.java

```diff
@@ -35,233 +36,36 @@
 @RegisterPlugin
 @Produces(CaveFacet.class)
 @Requires(@Facet(SurfaceHeightFacet.class))
-public class CaveFacetProvider implements ConfigurableFacetProvider, FacetProviderPlugin {
-    private CaveFacetProviderConfiguration configuration = new CaveFacetProviderConfiguration();
+public class CaveFacetProvider implements FacetProviderPlugin {
 
-    Noise baseCaveNoise;
-    Noise baseFadeCaveNoise;
+    SubSampledNoise3D[] caveNoise = new SubSampledNoise3D[2];
 
     @Override
     public void setSeed(long seed) {
-        baseCaveNoise = new PerlinNoise(seed + 2);
-        baseFadeCaveNoise = new PerlinNoise(seed + 3);
+        for(int i=0; i<2; i++) {
+            Noise3D pNoise = new SimplexNoise(seed + 2 + i);
```

Comment:
I've done some inlining and renaming. I think this resolves all your issues with this area. I left the brownian noise uninlined partially for length reasons and partially because it has settings that might need adjusting (the persistence and, less importantly, lacunarity).